fe/fe-core/src/main/java/com/starrocks/privilege/PrivilegeManager.java

```diff
@@ -48,9 +50,16 @@ protected PrivilegeManager() {
         typeStringToId = new HashMap<>();
         typeToActionMap = new HashMap<>();
         userToPrivilegeCollection = new HashMap<>();
+        roleIdToPrivilegeCollection = new HashMap<>();
         userLock = new ReentrantReadWriteLock();
+        roleLock = new ReentrantReadWriteLock();
+        nextRoleId = 0;
     }
 
+    private Map<Long, RolePrivilegeCollection> roleIdToPrivilegeCollection;
+    @SerializedName(value = "r")
+    private long nextRoleId;
```

Comment:
Why not using GlobalStateMgr.getNextId()?

Reply:
done